serializers/shared/src/main/java/io/pravega/schemaregistry/serializer/shared/codec/Encoder.java

```diff
@@ -36,5 +36,5 @@
      * @param outputStream ByteArrayOutputStream where the encoded data should be written.
      * @throws IOException IOException can be thrown while reading from or writing to byte buffers.
      */
-    void encode(ByteBuffer data, ByteArrayOutputStream outputStream) throws IOException;
+    void encode(ByteBuffer data, OutputStream outputStream) throws IOException;
```

Comment:
Previously, the interface expected a specific type of `java.io.OutputStream`, namely `java.io.ByteArrayOutputStream`, which allows writing bytes to a byte array in memory and not to any other type of destination like a socket or a file. 

The change expands the scope significantly. Now, any type of output stream can be passed. Does it have any potential for invalid/unexpected usage?

Reply:
the output stream is supplied by the caller of the method and here the implementer simply needs to write data into the output stream. it doesnt care about its capacity constraints per se. 
so we dont really need the ByteArrayOutputStream.